src/index.ts

```diff
@@ -105,6 +114,15 @@ export class Probot {
         privateKey: (privateKey && privateKey.toString()) || undefined,
         id: Number(process.env.APP_ID),
         port: Number(process.env.PORT) || 3000,
+        // XXX This seems to be the default Node behavior when host is not given:
+        //
+        // https://nodejs.org/api/net.html#net_server_listen
+        // If host is omitted, the server will accept connections on the unspecified IPv6 address (::) when IPv6 is available, or the unspecified IPv4 address (0.0.0.0) otherwise.
+        // In most operating systems, listening to the unspecified IPv6 address (::) may cause the net.Server to also listen on the unspecified IPv4 address (0.0.0.0).
+        //
+        // So probably 0.0.0.0 is needed here for backward compatibility, but localhost sounds
+        // like a more sensible and secure default to me...
+        bindAddress: process.env.BIND_ADDRESS || "localhost",
```

Comment:
I agree, `localhost` is more sensible, but a breaking change as you mention. Let's default to `0.0.0.0` or the IPv6 equivalent for now. We can change the default behavior in v11